Generating Sphinx documentation fails with an error after recent Jinja update

Following the [user documentation workflow](https://nest-simulator.readthedocs.io/en/latest/documentation_workflow/user_documentation_workflow.html) to build the Sphinx documentation fails with the following error:
```
Notebook error:
UndefinedError in model_details/HillTononiModels.ipynb:
'nbformat.notebooknode.NotebookNode object' has no attribute 'tags'
make[3]: *** [doc/CMakeFiles/html.dir/build.make:72: doc/CMakeFiles/html] Error 2
make[2]: *** [CMakeFiles/Makefile2:566: doc/CMakeFiles/html.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:573: doc/CMakeFiles/html.dir/rule] Error 2
make: *** [Makefile:231: html] Error 2
```

When creating a Conda environment with `extras/conda-nest-simulator-dev.yml` Jinja is installed as a dependency. Jinja was recently updated to version 3.0, which breaks the Sphinx build. It is likely that this will make building the documentation on Read the Docs fail as well. 

Jinja 3.0 support will be added with Sphinx 4.1.0, but it is still not released. Using any Jinja version before version 3.0 fixes the problem, so this requirement should be added to the yml file, and possibly also to `extras/nest-simulator-doc-requirements.txt` for RTD.

--- a/extras/conda-nest-simulator-dev.yml
+++ b/extras/conda-nest-simulator-dev.yml
@@ -32,12 +32,12 @@
 name: nest-simulator-dev
 channels:
   - conda-forge
-  
+
 dependencies:
   # Basics ------------------
   - python >= 3.8
   - pip
-  
+
   # Building NEST -----------
   - cmake >= 3.12
   - cython
@@ -78,6 +78,7 @@ dependencies:
   - sphinx-tabs
   - sphinx >= 2
   - sphinx_rtd_theme
+  - jinja2 < 3
   - tqdm
   - yamllint
 

--- a/extras/nest-simulator-doc-requirements.txt
+++ b/extras/nest-simulator-doc-requirements.txt
@@ -23,6 +23,7 @@ sphinx>=2
 sphinx_autobuild
 sphinx_gallery
 sphinx_rtd_theme
+jinja2<3
 statsmodels
 tqdm
 yamllint

